src/smt/term_formula_removal.cpp

```diff
@@ -50,14 +50,77 @@ RemoveTermFormulas::~RemoveTermFormulas() {}
 theory::TrustNode RemoveTermFormulas::run(
     Node assertion,
     std::vector<theory::TrustNode>& newAsserts,
-    std::vector<Node>& newSkolems)
+    std::vector<Node>& newSkolems,
+    bool fixedPoint)
 {
   Node itesRemoved = runInternal(assertion, newAsserts, newSkolems);
+  Assert(newAsserts.size() == newSkolems.size());
+  if (itesRemoved == assertion)
+  {
+    return theory::TrustNode::null();
+  }
+  // if running to fixed point, we run each new assertion through the
+  // run lemma method
+  if (fixedPoint)
+  {
+    size_t i = 0;
+    std::unordered_set<Node, NodeHashFunction> processed;
+    while (i < newAsserts.size())
```

Comment:
why not a `for`?

Reply:
Because newAsserts is being added to in the loop.

Reply:
So the size will change? Having the condition for the `for` as you have for the `while` would amount to the same, no?

Reply:
i think its clearer this way, since it suggests that a fixed point is being run.